### PR TITLE
KOGITO-5764 Proposed changes to examples for Quarkus 2 

### PR DESCRIPTION
**Pull request by antncl**
Please see https://issues.redhat.com/browse/KOGITO-5764

This is my WIP changes for $subject, been having a few issues documented on the ticket above.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `main` branch!**

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Comment by antncl** [comment]
Replacing quarkus-kogito-bom with kogito-quarkus-bom for now as written about on [KOGITO-5764](https://issues.redhat.com/browse/KOGITO-5764)

**Review by ricardozanini — APPROVED** [review]
Thanks Ant!

**Review by danielezonca — COMMENTED** [review]
I think this PR sort of revert the previous one to enable users to specify a different platform. I'm ok to avoid explicit quarkus version and using a property, I'm referring to the introduction of `kogito-quarkus-bom` as additional dependency.

Have you tested if it is still possible to replace platform with a different artifact after this PR? I expect not because now each project has `kogito-quarkus-bom` and also the platform so even if user changes the platform using maven properties kogito-quarkus-bom will remain in the scope.
I saw the discussion on the previous PR but I still think that the only solution is using `kogito-quarkus-bom` as "platform" default value and let user replace with `quarkus-kogito-bom` using the property like I suggested in [my original comment](https://github.com/kiegroup/kogito-examples/pull/828#discussion_r683626085)

We would like to have for delelopment a "quarkus-kogito-bom-alt" version where kogito is Snapshot while Quarkus is final. A similar bom doesn't exist as platform so I would stay with `kogito-quarkus-bom` as single dependency and explain user how to replace with real platform bom

@ricardozanini @evacchi 
Is it clear what I mean? I don't see any other solution to preserve a single dependency and also let user replace it, otherwise we will need to manipulate examples pom before each release to "prepare" them for the users.

**Comment by antncl** [comment]
@danielezonca @ricardozanini would someone else mind picking up on this PR and suggesting some concrete changes? I realise this issue is a little more complex and I'm likely starting work on the ER1 build very soon

**Comment by ricardozanini** [comment]
@danielezonca I though we settled that on master we will have `kogito-quarkus-bom` being used, and on stable branch the CI would replace with the correct BOMs + Quarkus platform.

File: decisiontable-quarkus-example/pom.xml [hunk 1]
```diff
@@ -10,13 +10,19 @@
   <artifactId>decisiontable-quarkus-example</artifactId>
   <name>Kogito Example :: Decision Table - Quarkus</name>
   <properties>
-    <quarkus-plugin.version>2.1.0.Final</quarkus-plugin.version>
     <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
-    <quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
-    <quarkus.platform.version>2.1.0.Final</quarkus.platform.version>
+    <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
+    <quarkus.platform.version>${version.io.quarkus}</quarkus.platform.version>
```
Comment (rsynek) [inline decisiontable-quarkus-example/pom.xml:15]: If I understand it correctly, this BOM is released as a part of the platform, not the core Quarkus. How can it have the same version coming from `kogito-build-parent`, if the platform is released after this repository?

The timeline of releases is as follows:
quarkus -> kogito -> quarkus platform

If we want to depend on platform BOMs, we would have to release `kogito-examples` only after the Quarkus platform is released.

After we merge https://github.com/kiegroup/kogito-runtimes/pull/1559, this PR will break until the 2.2.0.Final Quarkus platform is released.

Please correct me if I am missing something.

Comment (antncl) [reply decisiontable-quarkus-example/pom.xml:15]: @rsynek this is just for main, on product branches we'll have static versions here and on the prod side in theory they should get replaced with redhat built platform versions.

**Comment by radtriste** [comment]
See https://issues.redhat.com/browse/KOGITO-5764?focusedId=22095852&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-22095852

**Pull request closed without merging**